frontend/src/metabase/visualizations/visualizations/Table.jsx

```diff
@@ -125,21 +126,38 @@ export default class Table extends Component {
         const { data } = this.state;
         const sort = getIn(card, ["dataset_query", "query", "order_by"]) || null;
         const isPivoted = settings["table.pivot"];
+        const isColumnsDisabled = (settings["table.columns"] || []).filter(f => f.enabled).length < 1;
         const TableComponent = isDashboard ? TableSimple : TableInteractive;
 
         if (!data) {
             return null;
         }
 
-        return (
-            // $FlowFixMe
-            <TableComponent
-                {...this.props}
-                data={data}
-                isPivoted={isPivoted}
-                sort={sort}
-            />
-        );
+        if (isColumnsDisabled) {
+            return (
+                <div className={"flex-full px1 pb1 text-centered flex flex-column layout-centered " + (isDashboard ? "text-slate-light" : "text-slate")}>
```

Comment:
There's a utility called 'classnames' that you can use here to make this kind of conditional class application a bit easier.  If you look at https://github.com/metabase/metabase/pull/6339/files#diff-5114dfd896a1b8c2a72dd4e397024b01R68 you'll see an example.